### DISCUSSION TITLE
How to test reply path for onion messages

### DISCUSSION THREAD

**Discussion by valentinewallace** [ACINQ/eclair#2227] [Developers]
Hi! I'm trying to test that LDK properly constructs `reply_path`s when sending onion messages. 

I'm wondering if there's a way to send an onion message to an eclair node such that it will trigger the node to respond using the message's `reply_path`? :)

**Comment by t-bast** [comment]
There's nothing automatic built-in, because there's no response that really makes sense to send, it will depend on the actual scenario using onion messages.

However, it is straight-forward to add, so I went ahead and created a branch for you that automatically responds to incoming onion messages that contain a `reply_path`: https://github.com/ACINQ/eclair/tree/respond-to-onion-message

Thanks for the cross-impl tests!

**Reply by valentinewallace** [reply]
Oh awesome, thank you! I'll give that a shot

**Comment by valentinewallace** [comment]
@t-bast I'm running 7797292dc2303440525decdacab4a0d15d0f9b4d and testing an onion message: 
Eclair A -> LDK -> Eclair B --(reply)--> LDK -> Eclair A

but LDK isn't getting the reply onion message. Also don't see "sending dummy response to onion message" in `eclair.log`. 

Running eclair command: 
```
./bin/eclair-cli -p polarpass sendonionmessage --recipientNode=026006be22a0f7dfbed1ade19710e07580dbd2b09c8d05fce36795b4941f408ec2 --intermediateNodes=03c51ecba86e2345688512e1a8a8462c59d32b917d3af51497a72e6dee8b470358 --replyPath=03c51ecba86e2345688512e1a8a8462c59d32b917d3af51497a72e6dee8b470358 --content=2b03ffffff
{
  "sent": true,
  "failureMessage": "No response"
}
```
Also tried with an reply path containing both the intermediate LDK node ID and Eclair A's node id.

Here are my logs for the receiving/replying node: https://pastebin.com/gg7Wj60e

Any ideas? :)

**Reply by t-bast** [reply]
Your node is not running https://github.com/ACINQ/eclair/commit/7797292dc2303440525decdacab4a0d15d0f9b4d @valentinewallace :)
The first log line says it all:

```
2022-04-08 16:11:33,233 INFO  fr.acinq.eclair.Setup - version=0.7.0 commit=a804905
```

This commit is eclair v0.7.0 without the custom changes to respond to onion messages.

**Reply by valentinewallace** [reply]
Ah, sorry about that @t-bast 😅 

Sooo I'm now running the right commit (logs here for proof ;) https://pastebin.com/mpmMRSsy) but as you can see it's getting `java.lang.IllegalStateException: No native library found: at /fr/acinq/secp256k1/jni/native/darwin-aarch64/libsecp256k1-jni.dylib`. 

I haven't seen this error come up in eclair's issue tracking. Doesn't seem to occur when built on the prior `0.7.0` commit. Let me know if you have any tips, otherwise I can try to cherry-pick the changes onto the `0.7.0` commit maybe.. 

Running on the M1 Mac btw.

**Reply by t-bast** [reply]
Hum that one is probably for @sstone, it's related to the move from bouncycastle to secp256k1 bindings...
An easy solution is indeed to just rebase the commit I gave you on top of 0.7.0, this should be trivial and should unblock you!

**Reply by sstone** [reply]
Yes sorry about that we removed fallback to bouncycastle when secp256k1 is not available but do not yet provide bindings for macos M1 (darwin-aarch64). For now unfortunately you'll have to either use an older version of eclair as Bastien suggested, or an Intel mac.
